python/test/unit/fem/test_dofmap.py

```diff
@@ -524,6 +523,69 @@ def test_triangle_dof_ordering(space_type):
             assert np.allclose(edges[0][i], j[i])
 
 
+@pytest.mark.parametrize('space_type', [
+    ("P", 1), ("P", 2), ("P", 3), ("P", 4),
+    ("N1curl", 1),
+    ("RT", 1), ("RT", 2), ("RT", 3), ("RT", 4),
+    ("BDM", 1),
+    ("N2curl", 1),
+])
+def test_triangle_dof_ordering_parallel(space_type):
+    """Checks that dofs on shared triangle edges match up"""
+    # Create a triangle mesh
+    from random import shuffle
+    N = 10
+    points = np.array([[x / 2, y / 2] for x in range(N) for y in range(N)])
+    cells = []
+    for x in range(N - 1):
+        for y in range(N - 1):
+            a = N * y + x
+            for cell in [[a, a + 1, a + N + 1], [a, a + N + 1, a + N]]:
+                # Put points in cell in a random order to test dof permuting
+                shuffle(cell)
```

Comment:
Do we ultimately want to support arbitrary ordering, or insist on anti-clockwise ordering?

Reply:
I've changed this to have the cells in anticlockwise order, but the points in the mesh give a random order